### PR TITLE
fix: Improve duplicate flow name handling and add comprehensive tests

### PR DESCRIPTION
**Pull request by Cristhianzl**
This pull request improves the handling of duplicate flow names in the backend API and adds comprehensive unit tests to ensure robustness. The changes include refining the logic for naming duplicate flows, addressing edge cases such as special characters, numeric suffixes, and gaps in numbering, and introducing tests to validate these scenarios.

### Improvements to duplicate flow name handling:

* **Refined regex for matching duplicate flow names**: Updated the regex pattern to ensure that only flow names matching the exact pattern `{flow.name} ({number})` are processed, preventing unintended matches when the original name ends with a number in parentheses. (`src/backend/base/langflow/api/v1/flows.py`, [src/backend/base/langflow/api/v1/flows.pyL86-R88](diffhunk://#diff-37c1756ce77874f858b4862cefff35a28b8d1d3ec6534381934b3db73782eb26L86-R88))
* **Ensured consistent naming for duplicate flows**: Added logic to handle cases where no existing numbered duplicates are found, consistently appending `(1)` to the original name. (`src/backend/base/langflow/api/v1/flows.py`, [src/backend/base/langflow/api/v1/flows.pyR98-R99](diffhunk://#diff-37c1756ce77874f858b4862cefff35a28b8d1d3ec6534381934b3db73782eb26R98-R99))

### Comprehensive unit tests for edge cases:

* **Basic duplicate flow naming**: Validated that duplicate flows are correctly numbered sequentially (e.g., "Test Flow", "Test Flow (1)", "Test Flow (2)"). (`src/backend/tests/unit/api/v1/test_rename_flow_to_save.py`, [src/backend/tests/unit/api/v1/test_rename_flow_to_save.pyR1-R169](diffhunk://#diff-428c71c36f3ef8406aa7dbbb8321ccdacdc90f23484a1c94f04ed08b6710f8caR1-R169))
* **Flows with numbers in original names**: Tested scenarios where the original flow name includes numeric suffixes, ensuring proper numbering (e.g., "Untitled document (7)", "Untitled document (7) (1)"). (`src/backend/tests/unit/api/v1/test_rename_flow_to_save.py`, [src/backend/tests/unit/api/v1/test_rename_flow_to_save.pyR1-R169](diffhunk://#diff-428c71c36f3ef8406aa7dbbb8321ccdacdc90f23484a1c94f04ed08b6710f8caR1-R169))
* **Special characters and regex patterns**: Verified that flow names containing special characters or regex patterns are correctly escaped and handled during duplication. (`src/backend/tests/unit/api/v1/test_rename_flow_to_save.py`, [src/backend/tests/unit/api/v1/test_rename_flow_to_save.pyR1-R169](diffhunk://#diff-428c71c36f3ef8406aa7dbbb8321ccdacdc90f23484a1c94f04ed08b6710f8caR1-R169))
* **Handling gaps in numbering**: Ensured that gaps in numbering are addressed by using `max(numbers) + 1` for the next duplicate name. (`src/backend/tests/unit/api/v1/test_rename_flow_to_save.py`, [src/backend/tests/unit/api/v1/test_rename_flow_to_save.pyR1-R169](diffhunk://#diff-428c71c36f3ef8406aa7dbbb8321ccdacdc90f23484a1c94f04ed08b6710f8caR1-R169))

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved flow name duplication handling to ensure new flows are named correctly, even when the original name contains numbers or special characters.
  * Prevented incorrect name matching when the original flow name ends with a number in parentheses.

* **Tests**
  * Added comprehensive tests to verify flow name duplication logic, including cases with numbers, special characters, and gaps in numbering.

### CODE REVIEW

File: src/backend/base/langflow/api/v1/flows.py [hunk 1]
```diff
@@ -83,14 +83,18 @@ async def _new_flow(
                 )
             ).all()
             if flows:
-                extract_number = re.compile(r"\((\d+)\)$")
+                # Only extract numbers from flows that match the exact pattern "{flow.name} ({number})"
+                # This prevents matching the original flow name if it ends with a number in parentheses
+                extract_number = re.compile(rf"^{re.escape(flow.name)} \((\d+)\)$")
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/api/v1/flows.py]: I don't understand this part. The first commend says it should only match names with number and the second says it should not match names with number in parentheses.

Comment (Cristhianzl) [reply src/backend/base/langflow/api/v1/flows.py]: I'll improve the comments and add some examples, but the logic is:
Scenario 1: Original flow name is simple like "My Flow"
Copies: "My Flow (1)", "My Flow (2)", etc.
Scenario 2: Original flow name already contains parentheses like "My Project (Version 1)"
Copies should be: "My Project (Version 1) (1)", "My Project (Version 1) (2)", etc.

Thanks.

**Review by mfortman11 — APPROVED** [review]
